qt/scientific_interfaces/Muon/ALCDataLoadingPresenter.cpp

```diff
@@ -463,5 +385,118 @@ bool ALCDataLoadingPresenter::isLoading() const { return m_loadingData; }
  */
 void ALCDataLoadingPresenter::cancelLoading() const { m_LoadingAlg->cancel(); }
 
+void ALCDataLoadingPresenter::handleInstrumentChanged(std::string instrument) {
+  // Clear path as instrument has changed
+  m_view->setPath(std::string{});
+
+  // Update instrument
+  m_view->setInstrument(instrument);
+
+  // User cannot load yet as path not set
+  m_view->enableLoad(false);
+
+  // Turn off auto add
+  m_view->enableRunsAutoAdd(false);
+  m_view->toggleRunsAutoAdd(false);
+}
+
+void ALCDataLoadingPresenter::handleManageDirectories() {
+  MantidQt::API::ManageUserDirectories::openManageUserDirectories();
+}
+
+/**
+ * The watched directory has been changed - update flag.
+ * @param path :: [input] Path to directory modified (not used)
+ */
+void ALCDataLoadingPresenter::updateDirectoryChangedFlag(const QString &path) {
+  Q_UNUSED(path); // just set the flag, don't need the path
+  m_directoryChanged = true;
+}
+
+void ALCDataLoadingPresenter::startWatching(bool watch) {
+  if (watch) {
+    // Get path to watch and add to watcher
+    const auto path = m_view->getPath();
+    m_watcher.addPath(QString::fromStdString(path));
+
+    // start a timer that executes every second
+    m_timerID = startTimer(1000);
+  } else {
+    // Check if watcher has a directory, then remove all
+    if (!m_watcher.directories().empty()) {
+      m_watcher.removePaths(m_watcher.directories());
+    }
+
+    // Stop timer
+    killTimer(m_timerID);
+
+    // Reset latest auto run number and was range
+    m_lastRunLoadedAuto = -2;
+    m_wasLastAutoRange = false;
+  }
+}
+
+/**
+ * This timer runs every second when we are watching a directory.
+ * If any changes have occurred in the meantime, reload.
+ * @param timeup :: [input] Qt timer event (not used)
+ */
+void ALCDataLoadingPresenter::timerEvent(QTimerEvent *timeup) {
+  Q_UNUSED(timeup); // Only have one timer so do not need this
+
+  // Check if there are changes to watched directory
+  if (m_directoryChanged.load()) {
+    // Need to add most recent file to add to list
+    ALCLatestFileFinder finder(m_view->getFirstFile());
+    const auto latestFile = finder.getMostRecentFile();
+
+    // Check if file found
+    if (latestFile.empty()) {
+      // Error message?
```

Comment:
comment

Reply:
Ah I forgot to come back to this, so I was unsure on whether if the file was not found to display an error and stop auto or continue watching. The current functionality is to continue watching and will continually try loading the latest file again and again until success as the flag for directory changed is not set to false here. Which sounds better to you?

Reply:
If both are possible its best to ask Steve